From f6d7a026d3506364b23e0a7989038743821a3375 Mon Sep 17 00:00:00 2001
From: Stefan Haller <stefan@haller-berlin.de>
Date: Sun, 14 Nov 2010 11:47:35 +0100
Subject: [PATCH 06/19] gitk: Avoid horizontal scrolling of the commit list on
 Mac

With two-finger scrolling on a track pad, horizontal scrolling
of the commit list is usually unintentional, so avoid it
completely. This means it's also no longer possible to scroll
horizontally with shift-mousewheel, but this seems like a
small price to pay.
---
 gitk-git/gitk | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/gitk-git/gitk b/gitk-git/gitk
index 8b5f039d4a..c1a96bdc74 100755
--- a/gitk-git/gitk
+++ b/gitk-git/gitk
@@ -2705,8 +2705,9 @@ proc makewindow {} {
                 allcanvs yview scroll $delta units
             }
             bindall <Shift-MouseWheel> {
-                set delta [expr {- (%D)}]
-                $canv xview scroll $delta units
+                # Intentionally left empty. This prevents the canvas from
+                # scrolling vertically for horizontal wheel events, e.g.
+                # when moving diagonally with two fingers on a trackpad.
             }
         }
     }
-- 
2.50.0

